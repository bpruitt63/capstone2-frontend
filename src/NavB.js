import React from 'react';
import './static/styles/NavB.css';
import { NavLink as NotNavLink } from 'react-router-dom';
import {
    Nav,
    Navbar,
    NavbarBrand,
    NavItem,
    NavLink,
    UncontrolledDropdown,
    DropdownToggle,
    DropdownMenu,
    DropdownItem
  } from 'reactstrap';


function NavB({username, isMobile}) {
     
    return (
        <div>
            <Navbar color='dark'>
                <div className='container'>
                    <NavbarBrand href='/' className='inactive'>
                        Boatey
                    </NavbarBrand>
                    {!isMobile &&
                        <Nav>
                            <NavItem>
                                <NavLink to='/forecast' 
                                        className='inactive' 
                                        activeClassName='active' 
                                        tag={NotNavLink} >
                                    Weather Forecast
                                </NavLink>
                            </NavItem>
                            <NavItem>
                                <NavLink to='/planner' 
                                        className='inactive' 
                                        activeClassName='active' 
                                        tag={NotNavLink} >
                                    Trip Planner
                                </NavLink>
                            </NavItem>
                                {!username && 
                                    <>
                                        <NavItem>
                                            <NavLink to='/register'
                                                    className='inactive' 
                                                    activeClassName='active' 
                                                    tag={NotNavLink} >
                                                Register
                                            </NavLink>
                                        </NavItem>
                                        <NavItem>
                                            <NavLink to='/login'
                                                    className='inactive' 
                                                    activeClassName='active' 
                                                    tag={NotNavLink} >
                                                Log In
                                            </NavLink>
                                        </NavItem>
                                    </>
                                }
                                {username &&
                                    <>
                                        <NavItem>
                                            <NavLink to='/profile'
                                                    className='inactive' 
                                                    activeClassName='active' 
                                                    tag={NotNavLink} >
                                                {username}
                                            </NavLink>
                                        </NavItem>
                                        <NavItem>
                                            <NavLink to='/logout'
                                                    className='inactive' 
                                                    activeClassName='active' 
                                                    tag={NotNavLink} >
                                                Log Out
                                            </NavLink>
                                        </NavItem>
                                    </>
                                }
                            </Nav>
                        }
                        {isMobile &&
                            <Nav>
                                <UncontrolledDropdown nav inNavbar>
                                    <DropdownToggle nav caret className='inactive'>Options</DropdownToggle>
                                    <DropdownMenu right>
                                        <DropdownItem>
                                            <NavItem>
                                                <NavLink href='/forecast'>Weather Forecast</NavLink>
                                            </NavItem>
                                        </DropdownItem>
                                        <DropdownItem>
                                            <NavItem>
                                                <NavLink href='/planner'>Trip Planner</NavLink>
                                            </NavItem>
                                        </DropdownItem>
                                        {!username && 
                                            <>
                                                <DropdownItem>
                                                    <NavItem>
                                                        <NavLink href='/register'>Register</NavLink>
                                                    </NavItem>
                                                </DropdownItem>
                                                <DropdownItem>
                                                    <NavItem>
                                                        <NavLink href='/login'>Log In</NavLink>
                                                    </NavItem>
                                                </DropdownItem>
                                            </>
                                        }
                                        {username &&
                                            <>
                                                <DropdownItem>
                                                    <NavItem>
                                                        <NavLink href='/profile'>{username}</NavLink>
                                                    </NavItem>
                                                </DropdownItem>
                                                <DropdownItem>
                                                    <NavItem>
                                                        <NavLink href='/logout'>Log Out</NavLink>
                                                    </NavItem>
                                                </DropdownItem>
                                            </>
                                        }
                                    </DropdownMenu>
                                </UncontrolledDropdown>
                            </Nav>
                        }
                </div>
            </Navbar>
        </div>
    );
};

export default NavB;