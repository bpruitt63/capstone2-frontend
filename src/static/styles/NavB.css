.dropdown-menu-right {
    right: 0;
    left: auto;
}

.dropdown-menu-right button li a {
    color: black;
}

.dropdown-menu-right button li a:hover {
    color: darkgray;
}

a.inactive {
    color: white !important;
}

a.active {
    color: lightslategray !important;
}

a.inactive:hover {
    color: lightslategray !important;
}

a.active:hover {
    color: white !important;
}

