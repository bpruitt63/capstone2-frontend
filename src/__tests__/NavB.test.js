import React from 'react';
import { render, screen } from '@testing-library/react';
import NavB from '../NavB';
import { BrowserRouter } from 'react-router-dom';

test('renders', () => {
    render(<BrowserRouter><NavB/></BrowserRouter>);
    expect(screen.getByText(/Boatey/)).toBeInTheDocument;
});

test('renders logged out', () => {
    render(<BrowserRouter><NavB isMobile={false}/></BrowserRouter>);
    expect(screen.getByText(/Log In/)).toBeInTheDocument;
});

test('renders logged out Mobile', () => {
    render(<BrowserRouter><NavB isMobile={true}/></BrowserRouter>);
    expect(screen.getByText(/Options/)).toBeInTheDocument;
    expect(screen.getByText(/Log In/)).not.toBeInTheDocument;
});

test('renders logged in', () => {
    render(<BrowserRouter><NavB isMobile={false} username={'test'}/></BrowserRouter>);
    expect(screen.getByText(/test/)).toBeInTheDocument;
});

test('renders logged in Mobile', () => {
    render(<BrowserRouter><NavB isMobile={true} username={'test'}/></BrowserRouter>);
    expect(screen.getByText(/Options/)).toBeInTheDocument;
    expect(screen.getByText(/test/)).not.toBeInTheDocument;
});

test('matches snapshot', () => {
    const {asFragment} = render(<BrowserRouter><NavB/></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});