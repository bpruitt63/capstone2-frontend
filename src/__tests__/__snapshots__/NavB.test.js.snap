// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="inactive navbar-brand"
          href="/"
        >
          Boatey
        </a>
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="inactive nav-link"
              href="/forecast"
            >
              Weather Forecast
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="inactive nav-link"
              href="/planner"
            >
              Trip Planner
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="inactive nav-link"
              href="/register"
            >
              Register
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="inactive nav-link"
              href="/login"
            >
              Log In
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
